detector/vuls2/vuls2.go

```diff
@@ -576,53 +581,40 @@ func collectAdvisories(e ecosystemTypes.Ecosystem, family string, dres detectTyp
 	return am
 }
 
-func toCvss(v vulnerabilityTypes.Vulnerability, da *models.DistroAdvisory) (v2.CVSSv2, v31.CVSSv31) {
-	cvss2 := func() v2.CVSSv2 {
-		for _, s := range v.Content.Severity {
-			if s.Type == severityTypes.SeverityTypeCVSSv2 && s.CVSSv2 != nil {
-				return *s.CVSSv2
+func toCvss(v vulnerabilityTypes.Vulnerability) (v2.CVSSv2, v31.CVSSv31, v40.CVSSv40) {
+	var (
+		cvss2 v2.CVSSv2
+		cvss3 v31.CVSSv31
+		cvss4 v40.CVSSv40
+	)
+
+	for _, s := range v.Content.Severity {
+		switch s.Type {
+		case severityTypes.SeverityTypeCVSSv2:
+			if cvss2.Vector == "" && s.CVSSv2 != nil {
+				cvss2 = *s.CVSSv2
 			}
-		}
-		return v2.CVSSv2{}
-	}()
-	cvss3 := func() v31.CVSSv31 {
-		for _, s := range v.Content.Severity {
-			if s.Type == severityTypes.SeverityTypeCVSSv31 && s.CVSSv31 != nil {
-				return *s.CVSSv31
-			}
-		}
-		for _, s := range v.Content.Severity {
-			if s.Type == severityTypes.SeverityTypeCVSSv30 && s.CVSSv30 != nil {
-				return v31.CVSSv31{
+		case severityTypes.SeverityTypeCVSSv30:
+			if cvss3.Vector == "" && s.CVSSv30 != nil {
+				cvss3 = v31.CVSSv31{
 					Vector:       s.CVSSv30.Vector,
 					BaseScore:    s.CVSSv30.BaseScore,
 					BaseSeverity: s.CVSSv30.BaseSeverity,
 				}
 			}
-		}
-		return v31.CVSSv31{}
-	}()
-
-	advisorySeverity := func() string {
```

Comment:
I believe that the severity of the advisory should be in DistroAdvisory, so I decided that this section is unnecessary.